packages/dev/serializers/src/USD/USDSerializer.ts

```diff
@@ -0,0 +1,696 @@
+
+/* Converted from https://github.com/mrdoob/three.js/blob/dev/examples/jsm/exporters/USDZExporter.js */
+
+export interface IUSDExportOptions {
+    modelName?: string;
+    quickLookCompatible?: boolean;
+    ar?: IUSDArOptions;
+}
+
+interface IUSDArOptions {
+    anchoring: {
+        type: "plane" | "point";
+    };
+    planeAnchoring: {
+        alignment: "horizontal" | "vertical";
+    };
+}
+
+/**
+ * Class for generating USD data from a Babylon scene.
+ */
+export class USDExport {
+    /**
+     * Float Rounding Precision
+     */
+    public static Precision: number = 7;
+
+    /**
+     * Default Camera export values
+     * TODO get more information on these values and how they relate to Babylon Cameras
+     */
+    public static FilmGauge: number = 35;
+    public static Focus: number = 10;
+
+    /**
+     * Export Scene as USDZ file.
+     * @param scene to export
+     * @param options TODO get more information on USDZ options
+     * @param autoDownload automatically download the file.
+     * @returns a promise with the data of the USDZ file.
+     */
+    public static async ExportAsBinaryZip(scene: Scene, options: IUSDExportOptions = {}, autoDownload: boolean = true) {
+        options = {
+            ar: {
+                anchoring: { type: "plane" },
+                planeAnchoring: { alignment: "horizontal" },
+            },
+            quickLookCompatible: false,
+            ...options,
+        };
+
+        const files: any = {};
+        const modelFileName = `${options?.modelName || "model"}.usda`;
+        files[modelFileName] = null;
+
+        let output: string = USDExport._BuildHeader();
+        output += USDExport._BuildSceneStart(options);
+
+        const materials: { [id: string]: Material } = {};
+        const textures: { [id: string]: Texture } = {};
+
+        const sharedMat: PBRMaterial = new PBRMaterial(`default.mat`, scene);
+
+        scene.meshes
+            .filter((mesh) => (mesh as any).geometry)
+            .forEach((mesh) => {
+                const material = mesh.material ?? sharedMat;
+
+                switch (material.getClassName()) {
+                    case "PBRMaterial":
+                        const geometryFileName = "geometries/Geometry_" + mesh.uniqueId + ".usda";
+                        if (!(geometryFileName in files)) {
+                            const meshObject = USDExport._BuildMeshObject(mesh);
+                            files[geometryFileName] = USDExport._BuildUSDFileAsString(meshObject);
+                        }
+
+                        if (!(material.uniqueId in materials)) {
+                            materials[material.uniqueId] = material;
+                        }
+
+                        let noopNode: boolean = false;
+                        if (mesh.parent && isNoopNode(mesh.parent, false)) {
+                            (mesh.parent as TransformNode).scaling.z = 1;
+                        }

```

Comment:
> yes if that is the case, and I need to check pretty much every mesh for negative scaling usd does not support negative scales.

It is then up to the user not to use negative scaling. Here, we do some special handling because of the rhs => lhs conversion nodes we add ourselves at load time, so it's something we can handle. But for other cases, I think the best thing to do is simply issue a warning saying that conversion to usdz is not supported if the mesh we're trying to convert has negative scaling.

> maybe we should just try converting an PBR scene from the playground examples

Yes, you should try to convert some gltf files, to test the code.

Reply:
I have done some gltfs already, I was thinking none gltf imported scenes would be the thing to test now!

And sounds good.